pkg/estimator/server/replica/replica.go

```diff
@@ -15,3 +27,67 @@ func NodeMaxAvailableReplica(node *corev1.Node, pods []*corev1.Pod, request core
 	}
 	return int32(ni.MaxReplicaDivided(request)), nil
 }
+
+// ListerWrapper is a wrapper which wraps the pod lister and replicaset lister.
+type ListerWrapper struct {
+	listcorev1.PodLister
+	listappsv1.ReplicaSetLister
+}
+
+// GetUnschedulablePodsOfWorkload will return how many unschedulable pods a workload derives.
+func GetUnschedulablePodsOfWorkload(unstructObj *unstructured.Unstructured, threshold time.Duration, listers *ListerWrapper) (int32, error) {
+	if threshold < 0 {
+		threshold = 0
+	}
+	unschedulable := 0
+	// Workloads could be classified into two types. The one is which owns ReplicaSet
+	// and the other is which owns Pod directly.
+	switch unstructObj.GetKind() {
+	case util.DeploymentKind:
+		deployment, err := helper.ConvertToDeployment(unstructObj)
+		if err != nil {
+			return 0, fmt.Errorf("failed to convert ReplicaSet from unstructured object: %v", err)
+		}
+		pods, err := listDeploymentPods(deployment, listers)
+		if err != nil {
+			return 0, err
+		}
+		for _, pod := range pods {
+			if podUnschedulable(pod, threshold) {
+				unschedulable++
+			}
+		}
+	default:
+		// TODO(Garrybest): add abstract workload
+		return 0, fmt.Errorf("kind(%s) of workload(%s) is not supported", unstructObj.GetKind(), klog.KObj(unstructObj).String())
```

Comment:
Actually we check it here according to https://github.com/karmada-io/karmada/pull/1380#discussion_r811104915.

Reply:
That's OK.